pkg/kubelet/kubelet.go

```diff
@@ -1553,7 +1553,7 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if !runnable.Admit || pod.DeletionTimestamp != nil || apiPodStatus.Phase == v1.PodFailed {
+	if !runnable.Admit || pod.DeletionTimestamp != nil || kl.podIsTerminated(pod) {
```

Comment:
It looks like there were some statements further down in this function that branched on `kl.podIsTerminated` - can we delete those if checks?

Reply:
Addressed.